client/group.js

```diff
@@ -0,0 +1,47 @@
+Group = function(router, options, parent) {
+  options = options || {};
+
+  if (options.prefix && !/^\/.*/.test(options.prefix)) {
+    var message = "group's prefix must start with '/'";
+    throw new Error(message);
+  }
+
+  this._router = router;
+  this.prefix = options.prefix || '';
+
+  this._middlewares = options.middlewares || [];
+  this._subscriptions = options.subscriptions || Function.prototype;
+
+  if (parent) {
+    this.prefix = parent.prefix + this.prefix;
+    this._middlewares = parent._middlewares.concat(this._middlewares);
+    this.parent = parent;
```

Comment:
May be we can define parent always. try this.

~~~js
this.parent = parent;
if(this.parent) {
   ...
}
~~~